rfcs/convergence/move-out-n-rename-makestyles.md

```diff
@@ -0,0 +1,92 @@
+
+But until it stays as a folder in Fluent UI repo there is no clarity about it for third parties.
+
+### Lack of docs
+
+Compared to other CSS-in-JS `makeStyles` does not have enough docs to explain APIs and design. While every other library has them 🙄 Currently there is no right place for them.
+
+### "Bad" naming
+
+It's something that was not considered initially, but `makeStyles` is also a function in [MUI](https://mui.com/styles/basics/) and any search queries will point to it.
+
+### Opportunities for partners
+
+Without strict dependency on Fluent UI v9 we can potentially involve more partners to be consumers of `makeStyles` as they will be able to try new styling engine even without consuming v9 components.
+
+### Build issues
+
+Currently there are two unsolved issues: "buildless builds", "build dependency hell" once v9 & other packages are linked and circular dependencies with other repositories.
+
+#### Buildless environment
+
+The problem is described in [microsoft/fluentui#18357](https://github.com/microsoft/fluentui/issues/18357):
+
+- we cannot dogfood our Babel plugin/Webpack loader as it breaks buildless environment: to start Storybook you will need to run `yarn build --to @fluentui/make-styles-webpack-loader`
+- even now to run tests it's required to build `@fluentui/jest-serializer-make-styles`
+
+#### Dependency issues
+
+For incremental adoption with Teams we are looking into building compat layer between `makeStyles` & Fela ([microsoft/fluentui#20611](https://github.com/microsoft/fluentui/pull/20611)). This requires to be have `makeStyles` core as a dependency of Fluent UI React Northstar.
+
+Existing issues:
+
+- missing aliases for v9 packages in v0 configs
+- different TypeScript configuration
+- breaks in releases due different release cadences (v0 uses Lerna & manual releases)
+
+#### Circular dependencies between repos
+
+Currently `@fluentui/react-icons` depends on `@fluentui/react-make-styles`. When we will need to perform major bump of `@fluentui/react-make-styles`:
+
+- try to bump `@fluentui/react-make-styles` to v10
+- `@fluentui/react-icons` still uses v10 (double dependency)
+- cannot bump `@fluentui/react-make-styles` in `@fluentui/react-icons` as v10 is not released yet 💥
+
+## Detailed Design or Proposal
+
+- Create new repository in Microsoft org and move related packages
+- Rename packages to use new name and NPM scope (`makeStyles` will not use Fluent UI scope)
+- Improve documentation, engage community & partners
+
+### Pros and Cons
+
+Pros: It solves all problems described above ✅
+
+Cons:
+
+- Another repository could have visibility/engagement issues for Fluent UI team, but we already have Tabster, `@fluentui/react-icons` and others.
+- Potential confusion in docs: `makeStyles` & Fluent UI docs will be in separate places
```

Comment:
Would you consider using storybook for makeStyles docs?

Reply:
As we need this to composition, yes.

Reply:
This con is really just a temporary con. Once we have the website we can always add a section about `makeStyles` even if it's coming from a separate repo.